Editor/Modules/TextureModule.cs

```diff
@@ -0,0 +1,108 @@
+                new PropertyDefinition {type = PropertyTypeUtil.FromCustom(TextureProperties.Format), format = PropertyFormat.String, name = "Format", longName = "Texture Format" },
+                new PropertyDefinition { type = PropertyTypeUtil.FromCustom(TextureProperties.TextureCompression), format = PropertyFormat.String, name = "Compression", longName = "Texture Compression" },
+                new PropertyDefinition { type = PropertyTypeUtil.FromCustom(TextureProperties.MipMapEnabled), format = PropertyFormat.Bool, name = "MipMaps", longName = "Texture MipMaps Enabled" },
+                new PropertyDefinition { type = PropertyTypeUtil.FromCustom(TextureProperties.Readable), format = PropertyFormat.Bool, name = "Readable", longName = "Readable" },
+                new PropertyDefinition { type = PropertyTypeUtil.FromCustom(TextureProperties.Resolution), format = PropertyFormat.String, name = "Resolution", longName = "Texture Resolution" },
+                new PropertyDefinition { type = PropertyTypeUtil.FromCustom(TextureProperties.SizeOnDisk), format = PropertyFormat.Bytes, name = "Size", longName = "Texture Size" },
+                new PropertyDefinition { type = PropertyType.Path, name = "Path"}
+            }
+        };
+        public override string name => "Textures";
+
+        public override IReadOnlyCollection<IssueLayout> supportedLayouts => new IssueLayout[]
+        {
+            k_TexturesIssueLayout,
+        };
+
+
+        public override void Audit(ProjectAuditorParams projectAuditorParams, IProgress progress = null)
+        {
+            var allTextures = AssetDatabase.FindAssets("t: Texture, a:assets");
+            var issues = new List<ProjectIssue>();
+            var currentPlatform = projectAuditorParams.platform;
+
+            progress?.Start("Finding Textures", "Search in Progress...", allTextures.Length);
+
+            foreach (var aTexture in allTextures)
+            {
+                var pathToTexture = AssetDatabase.GUIDToAssetPath(aTexture);
+                var location = new Location(pathToTexture);
+
+                //next 2 lines: Grabs actual texture for inspecting its properties & continues if the object found is not a member of the Texture Group:(Texture2D, Texture3D, CubeMap, 2D Array)
+                var t = AssetImporter.GetAtPath(pathToTexture) as TextureImporter;
+                if (t == null) { continue; }
+
+                var tName = (Texture)AssetDatabase.LoadAssetAtPath(pathToTexture, typeof(Texture));
+                var tSize = Profiler.GetRuntimeMemorySizeLong(tName);
+                var platformSettings = t.GetPlatformTextureSettings(currentPlatform.ToString());
+
+                #if UNITY_2021_2_OR_NEWER
+                var resolution =  t.GetSourceTextureWidthAndHeight.width + "x" + t.GetSourceTextureWidthAndHeight.height,             //Not avail before Unity 2021.2
```

Comment:
I am pretty sure this one does not compile

Reply:
Strangely enough, the code doesn't compile in Unity 2021.3.6f1. 

What does compile is the code for the previous versions of Unity, so I will simply use that code for all editor versions, at this time. Updated.